orttraining/orttraining/training_api/optimizer.cc

```diff
@@ -85,33 +77,71 @@ std::unique_ptr<OptimizerAlgorithmBase> OptimizerAlorithmFactory::CreateInstance
         opt_type_to_freq_map[domain_type_pair] += 1;
       }
     }
-  }
+  } else {
 #else
-  // TODO (baijumeswani): Figure out the best way to extract the optimizer type
-  // from the model (either onnx model or ort format model) or from the checkpoint.
-  // For now, assume that the optimizer type is AdamWOptimizer in a minimal build.
-  ORT_UNUSED_PARAMETER(optim_path);
-
-  opt_type_to_freq_map[std::make_pair(kMSDomain, "AdamWOptimizer")] = 1;
+  ORT_UNUSED_PARAMETER(model);
+#endif
+    // TODO (baijumeswani): Figure out the best way to extract the optimizer type
+    // from the model (either onnx model or ort format model) or from the checkpoint.
+    // For now, assume that the optimizer type is AdamWOptimizer when using ort format models.
+    opt_type_to_freq_map[std::make_pair(kMSDomain, "AdamWOptimizer")] = 1;
+#if !defined(ORT_MINIMAL_BUILD)
+  }
 #endif
 
   ORT_ENFORCE(opt_type_to_freq_map.size() == 1U, "Only support one type of optimizer algorithm, but got: " +
                                                      std::to_string(opt_type_to_freq_map.size()));
   auto opt_it = opt_type_to_freq_map.begin();
   group_count = opt_it->second;
-  auto& domain = opt_it->first.first;
-  auto& type = opt_it->first.second;
+  auto& op_type = opt_it->first.second;
 
   // TODO: to support multiple groups, need to create a mapping between each group to its parameter list.
-  if (domain == kMSDomain && type == "AdamWOptimizer") {
+  if (op_type == "AdamWOptimizer") {
     return std::make_unique<AdamWOptimizerAlgorithm>();
-  } else if (domain == kMSDomain && type == "SGDOptimizerV2") {
+  } else if (op_type == "SGDOptimizerV2") {
     return std::make_unique<SGDOptimizerV2Algorithm>();
   } else {
     ORT_NOT_IMPLEMENTED("Not implemented for optimizer algo: " + opt_it->first.second);
   }
 }
 
+std::unique_ptr<OptimizerAlgorithmBase> OptimizerAlorithmFactory::CreateInstance(
+    const PathString& optim_path, int32_t& group_count) {
+  std::shared_ptr<Model> model = nullptr;
+#if !defined(ORT_MINIMAL_BUILD)
```

Comment:
is this check still needed since its caller already checked
```suggestion

```

Reply:
It is needed. The caller of this does not check, also not all the utilities are available in minimal build so a lot of times, the branched code is not valid in minimal build, that is why we need to simply remove it.